src/Features/CSharp/Portable/QuickInfo/CSharpSyntacticQuickInfoProvider.cs

```diff
@@ -28,18 +29,30 @@ public CSharpSyntacticQuickInfoProvider()
         protected override Task<QuickInfoItem?> BuildQuickInfoAsync(
             QuickInfoContext context,
             SyntaxToken token)
-            => Task.FromResult(BuildQuickInfo(token));
+            => Task.FromResult(BuildQuickInfo(token, context.CancellationToken));
 
         protected override Task<QuickInfoItem?> BuildQuickInfoAsync(
             CommonQuickInfoContext context,
             SyntaxToken token)
-            => Task.FromResult(BuildQuickInfo(token));
+            => Task.FromResult(BuildQuickInfo(token, context.CancellationToken));
 
-        private static QuickInfoItem? BuildQuickInfo(SyntaxToken token)
+        private static QuickInfoItem? BuildQuickInfo(SyntaxToken token, CancellationToken cancellationToken)
         {
-            if (token.Kind() != SyntaxKind.CloseBraceToken)
+            switch (token.Kind())
             {
-                return null;
+                case SyntaxKind.CloseBraceToken:
+                    break;
+                case SyntaxKind.EndRegionKeyword:
+                case SyntaxKind.HashToken:
+                case SyntaxKind.EndOfDirectiveToken:
+                    return BuildQuickInfoEndRegion(token, cancellationToken);
+                case SyntaxKind.EndOfFileToken:
+                    var endRegionKeyword = token.LeadingTrivia.LastOrDefault().GetStructure()?.ChildTokens().FirstOrDefault(t => t.IsKind(SyntaxKind.EndRegionKeyword));
+                    return endRegionKeyword.HasValue
+                        ? BuildQuickInfoEndRegion(endRegionKeyword.Value, cancellationToken)
+                        : null;
```

Comment:
I'm not sure if the EOF case is worth to handle. If so, we might try to find something better than `token.LeadingTrivia.LastOrDefault().GetStructure()?.ChildTokens().FirstOrDefault(t => t.IsKind(SyntaxKind.EndRegionKeyword))`